From 70957d2b60f5121582432ca28861daf5153c2182 Mon Sep 17 00:00:00 2001
From: Rajat Soni <quic_rajson@quicinc.com>
Date: Mon, 23 Jan 2023 11:07:40 +0530
Subject: [PATCH] ath12k: Add HTT_TX_PDEV_MU_MIMO_SCH_STATS_TLV stats

Some stats of HTT_TX_PDEV_MU_MIMO_SCH_STATS_TLV is missing.
Add HTT_TX_PDEV_MU_MIMO_SCH_STATS_TLV mu_mimo_sch_posted_per_group_index
values.

Signed-off-by: Rajat Soni <quic_rajson@quicinc.com>
---
 .../wireless/ath/ath12k/debugfs_htt_stats.c   | 71 ++++++++++++++++++-
 .../wireless/ath/ath12k/debugfs_htt_stats.h   | 11 +++
 2 files changed, 80 insertions(+), 2 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/debugfs_htt_stats.c b/drivers/net/wireless/ath/ath12k/debugfs_htt_stats.c
index 078e03c..e4a319a 100644
--- a/drivers/net/wireless/ath/ath12k/debugfs_htt_stats.c
+++ b/drivers/net/wireless/ath/ath12k/debugfs_htt_stats.c
@@ -1850,7 +1850,42 @@ htt_print_tx_pdev_mu_mimo_sch_stats_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "mu_mimo_ppdu_posted = %u\n",
 			 htt_stats_buf->mu_mimo_ppdu_posted);
 
-	len += scnprintf(buf + len, buf_len - len, "11ac MU_MIMO SCH STATS:\n");
+	len += scnprintf(buf + len, buf_len - len,
+			 "\nac_mu_mimo_sch_posted_per_group_index %u (SU) = %u,\n",
+			 0, htt_stats_buf->ac_mu_mimo_sch_posted_per_grp_sz[0]);
+	for (i = 1; i < HTT_TX_PDEV_STATS_NUM_AC_MUMIMO_USER_STATS; i++) {
+		len += scnprintf(buf + len, buf_len - len,
+		 "ac_mu_mimo_sch_posted_per_group_index %u (TOTAL STREAMS = %u) = %u,\n",
+		 i, i+1, htt_stats_buf->ac_mu_mimo_sch_posted_per_grp_sz[i]);
+	}
+
+	for (i = 0; i < HTT_TX_PDEV_STATS_NUM_AC_MUMIMO_USER_STATS; i++) {
+		len += scnprintf(buf + len, buf_len - len,
+		 "ac_mu_mimo_sch_posted_per_group_index %u (TOTAL STREAMS = %u) = %u,\n",
+		 i+HTT_TX_PDEV_STATS_NUM_AC_MUMIMO_USER_STATS,
+		 i+HTT_TX_PDEV_STATS_NUM_AC_MUMIMO_USER_STATS+1,
+		 htt_stats_buf->ac_mu_mimo_sch_posted_per_grp_sz_ext[i]);
+	}
+
+	len += scnprintf(buf + len, buf_len - len,
+		"\nax_mu_mimo_sch_posted_per_group_index %u (SU) = %u,\n",
+		0, htt_stats_buf->ax_mu_mimo_sch_posted_per_grp_sz[0]);
+	for (i = 1; i < HTT_TX_PDEV_STATS_NUM_AX_MUMIMO_USER_STATS; i++) {
+		len += scnprintf(buf + len, buf_len - len,
+		 "ax_mu_mimo_sch_posted_per_group_index %u (TOTAL STREAMS = %u) = %u,\n",
+		 i, i+1, htt_stats_buf->ax_mu_mimo_sch_posted_per_grp_sz[i]);
+	}
+
+	len += scnprintf(buf + len, buf_len - len,
+		"\nbe_mu_mimo_sch_posted_per_group_index %u (SU) = %u,\n",
+		0, htt_stats_buf->be_mu_mimo_sch_posted_per_grp_sz[0]);
+	for (i = 1; i < HTT_TX_PDEV_STATS_NUM_BE_MUMIMO_USER_STATS; i++) {
+		len += scnprintf(buf + len, buf_len - len,
+		 "be_mu_mimo_sch_posted_per_group_index %u (TOTAL STREAMS = %u) = %u,\n",
+		 i, i+1, htt_stats_buf->be_mu_mimo_sch_posted_per_grp_sz[i]);
+	}
+
+	len += scnprintf(buf + len, buf_len - len, "\n11ac MU_MIMO SCH STATS:\n");
 
 	for (i = 0; i < HTT_TX_PDEV_STATS_NUM_AC_MUMIMO_USER_STATS; i++)
 		len += scnprintf(buf + len, buf_len - len,
@@ -1864,12 +1899,44 @@ htt_print_tx_pdev_mu_mimo_sch_stats_tlv(const void *tag_buf,
 				 "ax_mu_mimo_sch_nusers_%u = %u\n",
 				 i, htt_stats_buf->ax_mu_mimo_sch_nusers[i]);
 
+	len += scnprintf(buf + len, buf_len - len, "\n11be MU_MIMO SCH STATS:\n");
+	for (i = 0; i < HTT_TX_PDEV_STATS_NUM_BE_MUMIMO_USER_STATS; i++) {
+		len += scnprintf(buf + len, buf_len - len,
+			"be_mu_mimo_sch_nusers_%u = %u\n",
+			i, htt_stats_buf->be_mu_mimo_sch_nusers[i]);
+	}
+
 	len += scnprintf(buf + len, buf_len - len, "\n11ax OFDMA SCH STATS:\n");
 
-	for (i = 0; i < HTT_TX_PDEV_STATS_NUM_OFDMA_USER_STATS; i++)
+	for (i = 0; i < HTT_TX_PDEV_STATS_NUM_OFDMA_USER_STATS; i++) {
 		len += scnprintf(buf + len, buf_len - len,
 				 "ax_ofdma_sch_nusers_%u = %u\n",
 				 i, htt_stats_buf->ax_ofdma_sch_nusers[i]);
+		len += scnprintf(buf + len, buf_len - len,
+				 "ax_ul_ofdma_basic_sch_nusers_%u = %u\n",
+				 i, htt_stats_buf->ax_ul_ofdma_basic_sch_nusers[i]);
+		len += scnprintf(buf + len, buf_len - len,
+				 "ax_ul_ofdma_bsr_sch_nusers_%u = %u\n",
+				 i, htt_stats_buf->ax_ul_ofdma_bsr_sch_nusers[i]);
+		len += scnprintf(buf + len, buf_len - len,
+				 "ax_ul_ofdma_bar_sch_nusers_%u = %u\n",
+				 i, htt_stats_buf->ax_ul_ofdma_bar_sch_nusers[i]);
+		len += scnprintf(buf + len, buf_len - len,
+				 "ax_ul_ofdma_brp_sch_nusers_%u = %u\n",
+				 i, htt_stats_buf->ax_ul_ofdma_brp_sch_nusers[i]);
+	}
+
+	len += scnprintf(buf + len, buf_len - len,
+			 "\n11ax UL MUMIMO SCH STATS:");
+
+	for (i = 0; i < HTT_TX_PDEV_STATS_NUM_UL_MUMIMO_USER_STATS; i++) {
+		len += scnprintf(buf + len, buf_len - len,
+			"ax_ul_mumimo_basic_sch_nusers_%u = %u\n", i,
+			htt_stats_buf->ax_ul_mumimo_basic_sch_nusers[i]);
+		len += scnprintf(buf + len, buf_len - len,
+			"ax_ul_mumimo_brp_sch_nusers_%u = %u\n", i,
+			htt_stats_buf->ax_ul_mumimo_brp_sch_nusers[i]);
+	}
 
 	stats_req->buf_len = len;
 }
diff --git a/drivers/net/wireless/ath/ath12k/debugfs_htt_stats.h b/drivers/net/wireless/ath/ath12k/debugfs_htt_stats.h
index 5ebc892..3209c41 100644
--- a/drivers/net/wireless/ath/ath12k/debugfs_htt_stats.h
+++ b/drivers/net/wireless/ath/ath12k/debugfs_htt_stats.h
@@ -972,6 +972,17 @@ struct htt_tx_pdev_mu_mimo_sch_stats_tlv {
 	u32 ac_mu_mimo_sch_nusers[HTT_TX_PDEV_STATS_NUM_AC_MUMIMO_USER_STATS];
 	u32 ax_mu_mimo_sch_nusers[HTT_TX_PDEV_STATS_NUM_AX_MUMIMO_USER_STATS];
 	u32 ax_ofdma_sch_nusers[HTT_TX_PDEV_STATS_NUM_OFDMA_USER_STATS];
+	u32 ax_ul_ofdma_basic_sch_nusers[HTT_TX_PDEV_STATS_NUM_OFDMA_USER_STATS];
+	u32 ax_ul_ofdma_bsr_sch_nusers[HTT_TX_PDEV_STATS_NUM_OFDMA_USER_STATS];
+	u32 ax_ul_ofdma_bar_sch_nusers[HTT_TX_PDEV_STATS_NUM_OFDMA_USER_STATS];
+	u32 ax_ul_ofdma_brp_sch_nusers[HTT_TX_PDEV_STATS_NUM_OFDMA_USER_STATS];
+	u32 ax_ul_mumimo_basic_sch_nusers[HTT_TX_PDEV_STATS_NUM_UL_MUMIMO_USER_STATS];
+	u32 ax_ul_mumimo_brp_sch_nusers[HTT_TX_PDEV_STATS_NUM_UL_MUMIMO_USER_STATS];
+	u32 ac_mu_mimo_sch_posted_per_grp_sz[HTT_TX_PDEV_STATS_NUM_AC_MUMIMO_USER_STATS];
+	u32 ax_mu_mimo_sch_posted_per_grp_sz[HTT_TX_PDEV_STATS_NUM_AX_MUMIMO_USER_STATS];
+	u32 be_mu_mimo_sch_nusers[HTT_TX_PDEV_STATS_NUM_BE_MUMIMO_USER_STATS];
+	u32 be_mu_mimo_sch_posted_per_grp_sz[HTT_TX_PDEV_STATS_NUM_BE_MUMIMO_USER_STATS];
+	u32 ac_mu_mimo_sch_posted_per_grp_sz_ext[HTT_TX_PDEV_STATS_NUM_AC_MUMIMO_USER_STATS];
 };
 
 struct htt_tx_pdev_mumimo_grp_stats_tlv {
-- 
2.17.1

